### ISSUE TITLE
Similar results obtained with Cross Entropy & HXE loss functions

### ISSUE THREAD

**Issue by coder-magpie** [fiveai/making-better-mistakes#8]
Hi Fiveai team,

Hope you are doing well.

I'm creating this issue to request your help on the HXE loss performance. I'm performing an experiment for research on the Animal dataset with the HXE loss function provided in the making better mistakes paper. I used both ResNet18 and ResNet50 architecture, with pre-trained ImageNet weights and the same configurations specified in the paper on a custom animal dataset. With multiple configurations, I observed both Cross-Entropy and HXE have very similar results. The top-1 accuracy, recall, f1-score, and hdist mistakes follow the same patterns. Could you please suggest what could have been wrong here, while using this loss function on the custom dataset? Also, the  HXE loss function considers all the nodes on the leaf level, but some classes could be non-leaf. I have a use case where one node is the parent of the other node, like genus-species, in such case distance between classes is different. How could we address this issue with the current implementation?

Looking forward to your reply.

Thanks,

**Comment by rhomu** [comment]
Thank you for your comment!

The HXE should work fine with non-leaf nodes... If you get results that are similar to XE you could try to change the parameter beta in order to make the loss more "aware" of the hierarchy?

**Comment by rhomu** [comment]
Closing for lack of activity.
